src/boot/bootctl.c

```diff
@@ -1062,20 +1122,22 @@ static int parse_argv(int argc, char *argv[]) {
                 ARG_NO_VARIABLES,
                 ARG_NO_PAGER,
                 ARG_GRACEFUL,
+                ARG_MAKE_MACHINE_ID_DIRECTORY,
         };
 
         static const struct option options[] = {
-                { "help",            no_argument,       NULL, 'h'                 },
-                { "version",         no_argument,       NULL, ARG_VERSION         },
-                { "esp-path",        required_argument, NULL, ARG_ESP_PATH        },
-                { "path",            required_argument, NULL, ARG_ESP_PATH        }, /* Compatibility alias */
-                { "boot-path",       required_argument, NULL, ARG_BOOT_PATH       },
-                { "print-esp-path",  no_argument,       NULL, 'p'                 },
-                { "print-path",      no_argument,       NULL, 'p'                 }, /* Compatibility alias */
-                { "print-boot-path", no_argument,       NULL, 'x'                 },
-                { "no-variables",    no_argument,       NULL, ARG_NO_VARIABLES    },
-                { "no-pager",        no_argument,       NULL, ARG_NO_PAGER        },
-                { "graceful",        no_argument,       NULL, ARG_GRACEFUL        },
+                { "help",                      no_argument,       NULL, 'h'                           },
+                { "version",                   no_argument,       NULL, ARG_VERSION                   },
+                { "esp-path",                  required_argument, NULL, ARG_ESP_PATH                  },
+                { "path",                      required_argument, NULL, ARG_ESP_PATH                  }, /* Compatibility alias */
+                { "boot-path",                 required_argument, NULL, ARG_BOOT_PATH                 },
+                { "print-esp-path",            no_argument,       NULL, 'p'                           },
+                { "print-path",                no_argument,       NULL, 'p'                           }, /* Compatibility alias */
+                { "print-boot-path",           no_argument,       NULL, 'x'                           },
+                { "no-variables",              no_argument,       NULL, ARG_NO_VARIABLES              },
+                { "no-pager",                  no_argument,       NULL, ARG_NO_PAGER                  },
+                { "graceful",                  no_argument,       NULL, ARG_GRACEFUL                  },
+                { "make-machine-id-directory", optional_argument, NULL, ARG_MAKE_MACHINE_ID_DIRECTORY },
```

Comment:
sorry for not noticing this earlier, but in new code we try not to use `optional_argument` anymore. It's a weird thing, because it creates ambiguities and we fell over them a couple of times. i.e. think:

```
foocmd --long-switch quux
```

Can mean two things: either `--long-switch` is passed with an argument of `quux` (i.e. equivalent to`--long-switch=quux`), or `--long-switch` is passed without argument, and a separate positional argument is later on passed as `quux`.

Hence, please just use `required_argument` here, then evertyhing is clear: an argument is expected and if there's none that's an error

Reply:
fair enough, I'd be lying if I said this didn't bite me in testing lol